sfs/mono/source.py

```diff
@@ -705,6 +705,101 @@ def plane_averaged_intensity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([i * n for n in n0])
 
 
+def pulsating_sphere_displacement(omega, x0, a, d, grid, c=None):
+    """Particle displacement of a pulsating sphere.
+
+    Paramters
+    ---------
+    omega : float
+        Frequency of pulsating sphere.
+    x0 : (3,) array_like
+        Position of source.
+    a : float
+        Radius of sphere.
+    d : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Particle displacement at positions given by *grid*.
+    """
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    offset = grid - x0
+    r = np.linalg.norm(offset)
+    r[r <= a] = np.nan
+    return util.XyzComponents([d * a / r**2 * np.exp(-1j * k * (r - a)) * o
+                               for o in offset])
+
+
+def pulsating_sphere_velocity(omega, x0, a, d, grid, c=None):
+    """Particle velocity of a pulsating sphere.
+
+    Paramters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    x0 : (3,) array_like
+        Position of source.
+    a : float
+        Radius of sphere.
+    d : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Particle velocity at positions given by *grid*.
+    """
+    return 1j * omega * pulsating_sphere_displacement(omega, x0, a, d, grid, c)
+
+
+def pulsating_sphere(omega, x0, a, d, grid, c=None):
+    """Sound pressure of a pulsating sphere.
+
+    Paramters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    x0 : (3,) array_like
+        Position of source.
+    a : float
+        Radius of sphere.
+    d : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Sound pressure at positions given by *grid*.
+    """
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    grid = util.as_xyz_components(grid)
+
+    r = np.linalg.norm(grid - x0)
+    r[r <= a] = np.nan
+    theta = np.arctan(1, k * r)
+    Z = defs.rho0 * defs.c * np.cos(theta) * np.exp(1j * theta)
+    return Z * 1j * omega * d * a / r * np.exp(-1j * k * (r - a))
```

Comment:
The return type is not a `XyzComponents`?

Reply:
Scalar fields should be `numpy.ndarray` not `XyzComponents`, right?

Reply:
Yes, of course you are right!

I was just confused because the resulting array was 1-dimensional, but that was just because your `grid` was broken (which is fixed in the meantime) and because the docstring was claiming that a `XyzComponents` is returned.

The same mistake was made in the `point()` and `plane()` functions, see #72.

Reply:
I also noticed that after your comment.
Actually I've copy-and-pasted from the `point()` function. I should have checked it.